src/repart/repart.c

```diff
@@ -5137,6 +5179,10 @@ static int context_copy_blocks(Context *context) {
         LIST_FOREACH(partitions, p, context->partitions) {
                 _cleanup_(partition_target_freep) PartitionTarget *t = NULL;
 
+                /* For offline signing case */
+                if (!set_isempty(arg_verity_settings) && IN_SET(p->type.designator, PARTITION_ROOT_VERITY_SIG, PARTITION_USR_VERITY_SIG))
+                        return partition_format_verity_sig(context, p);
+
```

Comment:
This shouldn't be done if the partition exists or is dropped or such. Please reorder the below conditions a bit and put this somewhere inbetween those